src/components/app/custom-els/FileDrop/index.ts

```diff
@@ -0,0 +1,141 @@
+import * as styles from './styles.css';
+import { bind } from '../../../../lib/util';
+
+
+function convertMimeTypeToRegex(mimeType: string): string {
+  return mimeType.replace(/\*/, '[^/]+' ).replace(/\//g, '\\/');
+}
+
+function hasMatchingFile(list: DataTransferItemList, accept: string): boolean {
+  const mimeRegex = convertMimeTypeToRegex(accept);
+  const acceptsRegex = new RegExp(mimeRegex);
+
+  const firstFile = Array.from(list).find(item => item.kind === 'file' && acceptsRegex.test(item.type))
+
+  return !!firstFile;
+}
+
+function firstMatchingFile(list: FileList, accept: string): File | undefined {
```

Comment:
You could probably make `hasMatchingFile` and `firstMatchingFile` the same function that takes `FileList | DataTransferItemList`.

Reply:
Yeah, I thought about that although the data structures are different and I had different logic...

Reply:
https://github.com/GoogleChromeLabs/squoosh/pull/56#discussion_r197836134 might help